pyccel/codegen/printing/ccode.py

```diff
@@ -1183,25 +1205,37 @@ def _print_OMP_Single_Construct(self, expr):
     def _print_Omp_End_Clause(self, expr):
         return '}'
     #=====================================
+    def free_allocs(self, variables):
+        allocs = set()
+        for var in variables:
+            for v in var:
+                allocs.add(v)
+        if len(allocs) > 0:
+            allocs_str = ', '.join(self._print(i.name) for i in allocs)
+            self._allocs = self._allocs[:len(self._allocs) - 2]
+            return 'free_allocs(' + str(len(allocs)) + ', ' + allocs_str + ');\n'
```

Comment:
Am I right in saying that here we right something like:
```
free_allocs(3, a, b, c);
```
?
It feels more natural to me to do:
```
free_allocs(a);
free_allocs(b);
free_allocs(c);
```
Do you have any reason for preferring the former?

Reply:
i just wanted the code to stay minimal that's all